pkg/cli/admin/inspectalerts/inspectalerts.go

```diff
@@ -0,0 +1,144 @@
+// Package inspectalerts provides access to in-cluster alerts.
+package inspectalerts
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/client-go/rest"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+
+	routev1 "github.com/openshift/api/route/v1"
+	routev1client "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+)
+
+// RouteGetter is a function that gets a Route.
+type RouteGetter func(ctx context.Context, namespace string, name string, opts metav1.GetOptions) (*routev1.Route, error)
+
+type Alert struct {
+}
+
+type options struct {
+	genericiooptions.IOStreams
+	RESTConfig *rest.Config
+	getRoute   RouteGetter
+}
+
+func newOptions(streams genericiooptions.IOStreams) *options {
+	return &options{
+		IOStreams: streams,
+	}
+}
+
+func New(f kcmdutil.Factory, streams genericiooptions.IOStreams) *cobra.Command {
+	o := newOptions(streams)
+	cmd := &cobra.Command{
+		Use:   "inspect-alerts",
+		Short: "Collect information about running alerts.",
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	return cmd
+}
+
+func (o *options) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	kcmdutil.RequireNoArguments(cmd, args)
+
+	cfg, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	o.RESTConfig = cfg
+
+	routeClient, err := routev1client.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	o.getRoute = func(ctx context.Context, namespace string, name string, opts metav1.GetOptions) (*routev1.Route, error) {
+		return routeClient.Routes(namespace).Get(ctx, name, opts)
+	}
+
+	return nil
+}
+
+func (o *options) Run(ctx context.Context) error {
+	alertBytes, err := GetAlerts(ctx, o.getRoute, o.RESTConfig.BearerToken)
```

Comment:
There are many ways to send authenticated request to API server in RESTConfig https://github.com/openshift/oc/blob/4fe33f165fbda5d9c3814dc4258859a502f43eb5/vendor/k8s.io/client-go/rest/config.go#L69-L89 and as far as I know, bearerToken is not mandatory. For example, if user may use credentials exec plugin which skips bearer token step.

Reply:
Bearer token is [what we doc][1], so it should be good enough for a env-var-gated first attempt, right?

[1]: https://docs.openshift.com/container-platform/4.14/monitoring/accessing-third-party-monitoring-apis.html#accessing-third-party-monitoring-web-service-apis_accessing-third-party-monitoring-apis

Reply:
Yes, I think so

Reply:
must-gather is [using a ServiceAccount token][1] and [collecting the default ingress CA][2].  Both of which we could use.  But :shrug:, also work to implement, and not clear to me that we need to sink that time in before landing a subcommand behind a `OC_ENABLE_CMD_INSPECT_ALERTS=true` gate.  Is "if there isn't a bearer token or the request is rejected with an HTTP auth code, go see if we have the power to borrow a must-gather ServiceAccount token" the sort of thing we need to land the initial implementation?  There's always work we could invest to make any implementation better, but where are the goal-posts for merging an initial, env-var-gated command?

[1]: https://github.com/openshift/must-gather/blob/c8fec85caeab4757d1c67daf38620fc89aabca69/collection-scripts/gather_monitoring#L27-L29
[2]: https://github.com/openshift/must-gather/blob/c8fec85caeab4757d1c67daf38620fc89aabca69/collection-scripts/gather_monitoring#L31-L34

Reply:
RE: stable API , we can take up the work post the merge. This should not block the merge as this PR is behind  OC_ENABLE_CMD_INSPECT_ALERTS=true flag.

Reply:
> Isn't there a stable API we could access? How our console is accessing that data? Can we re-use that same API?

The console uses the same API service to query and display the cluster alerts.